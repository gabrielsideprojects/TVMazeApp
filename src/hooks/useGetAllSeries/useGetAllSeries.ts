import {useInfiniteQuery, UseInfiniteQueryResult} from 'react-query';
import {api} from '../../api';
import {Serie} from '../../types/series';

async function getSeries(pageParam: number) {
  const result = await api.get(`shows?page=${pageParam}`);

  return result.data;
}

export function useGetAllSeries(): UseInfiniteQueryResult<Serie[]> {
  return useInfiniteQuery(
    ['allSeries'],
    async ({pageParam = 1}) => await getSeries(pageParam),
    {
      getNextPageParam: lastPage =>
        lastPage[0].id ? lastPage[0].id : undefined,
    },
  );
}
