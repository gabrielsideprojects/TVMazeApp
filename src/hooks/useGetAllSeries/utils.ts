import {InfiniteData} from 'react-query';
import {Serie} from '../../types/series';

export function getDataFromAllSeriesPages(
  list: InfiniteData<Serie[]> | undefined,
): Serie[] {
  if (!list) {
    return [];
  }
  return list.pages.reduce((prev, curr) => {
    return [...prev, ...curr];
  }, [] as Serie[]);
}
