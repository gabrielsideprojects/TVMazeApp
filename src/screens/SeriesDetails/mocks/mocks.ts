export const characterDetailsMock = {
    created: 'string',
    episode: ['teste'],
    gender: 'M',
    id: '123',
    image: 'https://url.com',
    location: {
        name: 'name',
        url: 'url'
    },
    name: 'Ricky',
    origin: {
        name: 'originName',
        url: 'originUrl'
    },
    species: 'specie',
    status: 'Status',
    type: 'Type',
    url: 'url'
}

export const episodeListMock = [{
    id: '123',
    name: 'Humberto',
    air_date: 'date',
    episode: '44',
    characters: ['Morty'],
    url: 'https://fakeUrl.com',
    created: 'created'
}]