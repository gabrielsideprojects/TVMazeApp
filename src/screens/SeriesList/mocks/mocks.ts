const characterMocked = {
    id: '789',
    name: 'Ricky',
    status: 'Morty',
    species: 'Alien',
    image: 'https://url.com',
    type: 'Human',
    gender: 'M',
    origin: {
      name: 'Name',
      url: 'url'
    },
    location: {
      name: 'Location',
      url: 'url'
    },
    episode: ['555', 'episode'],
    url: 'url',
    created: 'date'
}
const characterMockedWithDifferentId = {
  ...characterMocked,
  id: '452',
}

export const characterMock = {
    info: {
        count: 0,
        pages: 2,
        next: '3',
        prev: '2',
      },
      results: [characterMocked,characterMockedWithDifferentId]
}

