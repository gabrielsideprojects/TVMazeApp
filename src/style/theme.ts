export const lightTheme = {
  colors: {
    yellow: '#f0e14a',
    pink: '#e4a788',
    green: '#97ce4c',
    white: '#ffffff',
    black: '#000',
    gray: '#808080',
    blue: '#00008b',
    background: '#fff',
  },
  borderRadius: {
    small: 4,
    medium: 8,
    large: 16,
    extraLarge: 65,
  },
  spacing: {
    small: 4,
    medium: 8,
    semiLarge: 12,
    large: 16,
    extraLarge: 32,
  },
};

export const darkTheme = {
  colors: {
    yellow: '#f0e14a',
    pink: '#e4a788',
    green: '#97ce4c',
    white: '#fff',
    black: '#000',
    blue: '#00008b',
    gray: '#808080',
    background: '#000',
  },
  borderRadius: {
    small: 4,
    medium: 8,
    large: 16,
    extraLarge: 65,
  },
  spacing: {
    small: 4,
    medium: 8,
    semiLarge: 12,
    large: 16,
    extraLarge: 32,
  },
};
