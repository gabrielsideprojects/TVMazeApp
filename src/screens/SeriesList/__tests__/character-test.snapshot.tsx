import React from 'react';
import {CharacterScreen} from '..';
import {useGetAllCharacters} from '../../../hooks/useGetAllSeries/useGetAllSeries';
import {characterMock} from '../mocks/mocks';
import {render} from '@testing-library/react-native';
import {ThemeProvider} from 'styled-components/native';
import {theme} from '../../../style/theme';
// import { render } from '../../../utils/testUtils';

const mockedUseUsersQuery = useGetAllCharacters as jest.Mock<any>;

// Mock the hook module
jest.mock('../../../hooks/useGetAllCharacters/useGetAllCharacters');

describe('Character snapshot testing', () => {
  beforeEach(() => {
    mockedUseUsersQuery.mockImplementation(() => ({
      status: 'success',
      data: {
        pages: [characterMock],
      },
    }));
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should match snapshot', () => {
    const tree = render(
      <ThemeProvider theme={theme}>
        {' '}
        <CharacterScreen />{' '}
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
