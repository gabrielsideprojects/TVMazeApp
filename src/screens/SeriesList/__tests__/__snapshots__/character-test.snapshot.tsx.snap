// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character snapshot testing should match snapshot 1`] = `
Array [
  " ",
  <RCTSafeAreaView
    style={
      Array [
        Object {
          "paddingBottom": 32,
        },
      ]
    }
  >
    <RCTSafeAreaView
      style={
        Array [
          Object {
            "paddingBottom": 32,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "marginBottom": 16,
              "paddingBottom": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 0,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icon.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 30,
                  "width": 30,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Roboto-Bold",
                  "fontSize": 20,
                },
              ]
            }
          >
            Rick and Morty
          </Text>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î…’
            </Text>
          </View>
        </View>
      </View>
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        ListFooterComponent={null}
        data={
          Array [
            Object {
              "created": "date",
              "episode": Array [
                "555",
                "episode",
              ],
              "gender": "M",
              "id": "789",
              "image": "https://url.com",
              "location": Object {
                "name": "Location",
                "url": "url",
              },
              "name": "Ricky",
              "origin": Object {
                "name": "Name",
                "url": "url",
              },
              "species": "Alien",
              "status": "Morty",
              "type": "Human",
              "url": "url",
            },
            Object {
              "created": "date",
              "episode": Array [
                "555",
                "episode",
              ],
              "gender": "M",
              "id": "452",
              "image": "https://url.com",
              "location": Object {
                "name": "Location",
                "url": "url",
              },
              "name": "Ricky",
              "origin": Object {
                "name": "Name",
                "url": "url",
              },
              "species": "Alien",
              "status": "Morty",
              "type": "Human",
              "url": "url",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onEndReached={[Function]}
        onEndReachedThreshold={0.5}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#e4a788",
                    "borderRadius": 16,
                    "flexDirection": "row",
                    "marginBottom": 0,
                    "marginLeft": 8,
                    "marginRight": 8,
                    "marginTop": 0,
                    "paddingBottom": 8,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "borderRadius": 65,
                        "height": 60,
                        "width": 60,
                        "zIndex": 1,
                      },
                    ],
                  ]
                }
              >
                <FastImageView
                  defaultSource={null}
                  resizeMode="cover"
                  source={
                    Object {
                      "uri": "https://url.com",
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "justifyContent": "space-between",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                        "marginLeft": 16,
                      },
                    ]
                  }
                >
                  <Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "Roboto-Bold",
                          },
                        ]
                      }
                    >
                      #
                      789
                       
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "Roboto-Medium",
                            "marginLeft": 8,
                          },
                        ]
                      }
                    >
                      Ricky
                    </Text>
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                        "marginLeft": 16,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#f0e14a",
                          "borderRadius": 8,
                          "marginRight": 8,
                          "marginTop": 8,
                          "paddingBottom": 8,
                          "paddingLeft": 8,
                          "paddingRight": 8,
                          "paddingTop": 8,
                        },
                      ]
                    }
                  >
                    <Text>
                      Morty
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#f0e14a",
                          "borderRadius": 8,
                          "marginRight": 8,
                          "marginTop": 8,
                          "paddingBottom": 8,
                          "paddingLeft": 8,
                          "paddingRight": 8,
                          "paddingTop": 8,
                        },
                      ]
                    }
                  >
                    <Text>
                      Alien
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={false}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "marginRight": 8,
                        "marginTop": 8,
                        "opacity": 1,
                        "paddingBottom": 8,
                        "paddingLeft": 8,
                        "paddingRight": 8,
                        "paddingTop": 8,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      onPress={[Function]}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ï€„
                    </Text>
                  </View>
                </View>
              </View>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ï„…
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "height": 8,
                  },
                ]
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#e4a788",
                    "borderRadius": 16,
                    "flexDirection": "row",
                    "marginBottom": 0,
                    "marginLeft": 8,
                    "marginRight": 8,
                    "marginTop": 0,
                    "paddingBottom": 8,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "borderRadius": 65,
                        "height": 60,
                        "width": 60,
                        "zIndex": 1,
                      },
                    ],
                  ]
                }
              >
                <FastImageView
                  defaultSource={null}
                  resizeMode="cover"
                  source={
                    Object {
                      "uri": "https://url.com",
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "justifyContent": "space-between",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                        "marginLeft": 16,
                      },
                    ]
                  }
                >
                  <Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "Roboto-Bold",
                          },
                        ]
                      }
                    >
                      #
                      452
                       
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "Roboto-Medium",
                            "marginLeft": 8,
                          },
                        ]
                      }
                    >
                      Ricky
                    </Text>
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                        "marginLeft": 16,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#f0e14a",
                          "borderRadius": 8,
                          "marginRight": 8,
                          "marginTop": 8,
                          "paddingBottom": 8,
                          "paddingLeft": 8,
                          "paddingRight": 8,
                          "paddingTop": 8,
                        },
                      ]
                    }
                  >
                    <Text>
                      Morty
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#f0e14a",
                          "borderRadius": 8,
                          "marginRight": 8,
                          "marginTop": 8,
                          "paddingBottom": 8,
                          "paddingLeft": 8,
                          "paddingRight": 8,
                          "paddingTop": 8,
                        },
                      ]
                    }
                  >
                    <Text>
                      Alien
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={false}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "marginRight": 8,
                        "marginTop": 8,
                        "opacity": 1,
                        "paddingBottom": 8,
                        "paddingLeft": 8,
                        "paddingRight": 8,
                        "paddingTop": 8,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      onPress={[Function]}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ï€„
                    </Text>
                  </View>
                </View>
              </View>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ï„…
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </RCTSafeAreaView>
  </RCTSafeAreaView>,
  " ",
]
`;
